src/cogs/rules.py

```diff
@@ -56,6 +57,37 @@ def cleanup_code(self, content):
         # remove `foo`
         return content.strip('` \n')
 
+    async def _update_rules(self):
+        with open("src/backend/database.json") as file:
+            j = json.load(file)
+        
+        channel = self.bot.get_channel(j["rules_channel"])
+        messages = await channel.history(limit=1000000, oldest_first=True).flatten()
+        self._rules = {}
+
+        for message in messages:
+            content = message.clean_content
+            matches = re.finditer(r"(\d+) - (.+?)(?=[\n ]+\d+? - |$)", content, flags=re.DOTALL)
+
+            for rule in matches:
+                if rule[0] != "":
+                    number = rule[1]
+                    text = rule[2]
+
+                    if self._rules.get(number) is None:
+                        self._rules[number] = text
+
+    @commands.command()
+    async def update_rules(self, ctx):
+        def get_permitted():
+            with open("src/backend/database.json", 'r') as file:
+                return json.load(file)["permitted"]
```

Comment:
This funciton already exists elsewhere. import and use it.

Reply:
![image](https://user-images.githubusercontent.com/54042101/114244967-ca663100-998f-11eb-8937-54bfc036094a.png)
Hmmm

Also, it was only ever locally defined. So I can't access it.